### PR TITLE
{bio}[foss/2022a] Xmipp v3.22.07-Helios w/ Python 3.10.4

### PR DESCRIPTION
**Pull request by akesandgren**
(created using `eb --new-pr`)

depends on: ~~https://github.com/easybuilders/easybuild-easyblocks/pull/2809~~, https://github.com/easybuilders/easybuild-easyblocks/pull/2818

### CODE REVIEW

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
b-an02.hpc2n.umu.se - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 3.8.10
See https://gist.github.com/e4c1600a5427d0eda30546e08fadfa9c for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**FAILED**
Build succeeded for 1 out of 3 (3 easyconfigs in total)
b-an02.hpc2n.umu.se - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 3.8.10
See https://gist.github.com/859b94ae3c7760fe4702eced9505d23f for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
b-an02.hpc2n.umu.se - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 3.8.10
See https://gist.github.com/0ca936ee7b86ae9f88c48bc409950f9b for a full test report.

**Comment by branfosj** [comment]
@boegelbot please test @ jsc-zen2

**Comment by branfosj** [comment]
@boegelbot please test @ jsc-zen2

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 25 out of 25 (3 easyconfigs in total)
bear-pg0105u36b.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/6a0a7f98ba50c7f79a5d89af1157a8ca for a full test report.

File: easybuild/easyconfigs/x/Xmipp/Xmipp-3.22.07-Helios-foss-2022a.eb [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+builddependencies = [
+    ('SCons', '4.4.0'),
+    ('CTPL', '0.0.2'),
+]
+
+dependencies = [
+    ('Python', '3.10.4'),
+    ('SciPy-bundle', '2022.05'),
+    ('Java', '11', '', SYSTEM),
+    ('OpenCV', '4.6.0', '-contrib'),
```
Comment (branfosj) [inline easybuild/easyconfigs/x/Xmipp/Xmipp-3.22.07-Helios-foss-2022a.eb:79]:
```
^[[92m g++ -c -w -O2 -ftree-vectorize -march=native -fno-math-errno -std=c++17 xmipp_test_opencv.cpp -o xmipp_test_opencv.o -I../ -I/rds/projects/2017/branfosj-rse/easybuild/EL8-ice/software/Python/3.10.4-GCCcore-11.3.0/include^[[0m
^[[93m xmipp_test_opencv.cpp:1:10: fatal error: opencv2/core/core.hpp: No such file or directory    1 | #include <opencv2/core/core.hpp>      |          ^~~~~~~~~~~~~~~~~~~~~~~compilation terminated.^[[0m


^[[93m OpenCV not found^[[0m
```

**Review by branfosj — CHANGES_REQUESTED** [review]
It is looking for `OpenCV` and `HDF5` in Python?

File: easybuild/easyconfigs/x/Xmipp/Xmipp-3.22.07-Helios-foss-2022a.eb [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    ('SQLite', '3.38.3'),
+    ('HDF5', '1.12.2'),
```
[the remark this answers is not in the document]
Comment (akesandgren) [reply easybuild/easyconfigs/x/Xmipp/Xmipp-3.22.07-Helios-foss-2022a.eb:81]: Are you running with the latest xmipp.py from develop?

[the remark this answers is not in the document]
Comment (akesandgren) [reply easybuild/easyconfigs/x/Xmipp/Xmipp-3.22.07-Helios-foss-2022a.eb:81]: Or was that despite the build being successful?

[the remark this answers is not in the document]
Comment (akesandgren) [reply easybuild/easyconfigs/x/Xmipp/Xmipp-3.22.07-Helios-foss-2022a.eb:81]: Oh, I have the same OpenCV error, it's looking for opencv2 not opencv4...

[the remark this answers is not in the document]
Comment (akesandgren) [reply easybuild/easyconfigs/x/Xmipp/Xmipp-3.22.07-Helios-foss-2022a.eb:81]: The problem then is that OpenCV only sets CPATH to $EBROOTOPENCV/include, and doesn't add $EBROOTOPENCV/include/opencv4

[the remark this answers is not in the document]
Comment (akesandgren) [reply easybuild/easyconfigs/x/Xmipp/Xmipp-3.22.07-Helios-foss-2022a.eb:81]: https://github.com/easybuilders/easybuild-easyblocks/pull/2818 fixes the problem in OpenCV 4.x

[the remark this answers is not in the document]
Comment (akesandgren) [reply easybuild/easyconfigs/x/Xmipp/Xmipp-3.22.07-Helios-foss-2022a.eb:81]: With that I get:
```
OPENCV-4 detected without CUDA support
```

[the remark this answers is not in the document]
Comment (branfosj) [reply easybuild/easyconfigs/x/Xmipp/Xmipp-3.22.07-Helios-foss-2022a.eb:81]: Yes, I am getting `OPENCV-4 detected without CUDA support` as well now.

However, I am still seeing
```
Headers for 'libhdf5' not found at '/rds/projects/2017/branfosj-rse/easybuild/EL8-ice/software/Python/3.10.4-GCCcore-11.3.0/include'.
No alternative found in the system.
```

File: easybuild/easyconfigs/x/Xmipp/Xmipp-3.22.07-Helios-foss-2022a.eb (hunk 1 above)
Comment (branfosj) [reply easybuild/easyconfigs/x/Xmipp/Xmipp-3.22.07-Helios-foss-2022a.eb:79]: resolved by https://github.com/easybuilders/easybuild-easyblocks/pull/2818

File: easybuild/easyconfigs/x/Xmipp/Xmipp-3.22.07-Helios-foss-2022a.eb (hunk 2 above)
[the remark this answers is not in the document]
Comment (branfosj) [reply easybuild/easyconfigs/x/Xmipp/Xmipp-3.22.07-Helios-foss-2022a.eb:81]: Ah, so it complains about not finding the headers but finds the library and is fine.

**Comment by branfosj** [comment]
@boegelbot please test @ jsc-zen2

**Comment by branfosj** [comment]
@boegelbot please test @ generoso

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
bear-pg0105u36b.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/ca59b84e20ade9be754c4e076dcd868e for a full test report.

**Review by branfosj — APPROVED** [review]

**Comment by branfosj** [comment]
Going in, thanks @akesandgren!

**Pull request merged**